src/plugins/intel_cpu/src/infer_request.h

```diff
@@ -4,106 +4,94 @@
-#include <cpp_interfaces/interface/ie_iinfer_request_internal.hpp>
+
+#include "graph.h"
+#include "openvino/runtime/iinfer_request.hpp"
+#include "openvino/runtime/isync_infer_request.hpp"
 
 namespace ov {
 namespace intel_cpu {
 
-class ExecNetwork;
+class CompiledModel;
 class AsyncInferRequest;
 
-class InferRequestBase : public InferenceEngine::IInferRequestInternal {
+class SyncInferRequest : public ov::ISyncInferRequest {
 public:
-    virtual ~InferRequestBase();
+    SyncInferRequest(std::shared_ptr<const CompiledModel> compiled_model);
+    virtual ~SyncInferRequest();
+
+    void infer() override;
 
-    void InferImpl() override;
+    std::vector<ov::ProfilingInfo> get_profiling_info() const override;
 
-    std::map<std::string, InferenceEngine::InferenceEngineProfileInfo> GetPerformanceCounts() const override;
+    std::vector<std::shared_ptr<ov::IVariableState>> query_state() const override;
 
-    std::vector<std::shared_ptr<InferenceEngine::IVariableStateInternal>> QueryState() override;
+    void set_tensor(const ov::Output<const ov::Node>& port, const ov::Tensor& tensor) override;
+
+    void set_tensors_impl(const ov::Output<const ov::Node> port, const std::vector<ov::Tensor>& tensors) override;
+
+    ov::Tensor get_tensor(const ov::Output<const ov::Node>& port) const override;
+    std::vector<ov::Tensor> get_tensors(const ov::Output<const ov::Node>& _port) const override;
 
     /**
      * @brief      Sets the pointer to asynchronous inference request that holds this request
      * @param[in]  asyncRequest Pointer to asynchronous inference request
      */
-    void SetAsyncRequest(AsyncInferRequest* asyncRequest);
+    void set_async_request(AsyncInferRequest* asyncRequest);
 
     /**
-     * @brief If `_asyncRequest` is initialized throw exception with `InferenceEngine::INFER_CANCELLED` status if inference request is canceled
+     * @brief If `m_asyncRequest` is initialized throw exception with `ov::Cancelled` status if inference request is
+     * canceled
      */
-    void ThrowIfCanceled() const;
+    void throw_if_canceled() const;
 
 protected:
-    InferRequestBase(InferenceEngine::InputsDataMap networkInputs,
-                     InferenceEngine::OutputsDataMap networkOutputs,
-                     std::shared_ptr<ExecNetwork> execNetwork_)
-    : IInferRequestInternal(networkInputs, networkOutputs), execNetwork(execNetwork_) {}
-
-    InferRequestBase(const std::vector<std::shared_ptr<const ov::Node>>& inputs,
-                     const std::vector<std::shared_ptr<const ov::Node>>& outputs,
-                     std::shared_ptr<ExecNetwork> execNetwork_)
-    : IInferRequestInternal(inputs, outputs), execNetwork(execNetwork_) {}
+    void create_infer_request();
+    InferenceEngine::Precision norm_to_input_supported_prec(const std::pair<const std::string, ov::Tensor>& input) const;
+    void pushInput(const std::string& inputName, ov::Tensor& inputBlob, InferenceEngine::Precision dataType);
```

Comment:
Can we merge pushInput into push_input_data to simplify the class? And I cannot see why pushInput needs the 3rd parameter dataType. There is already a cpu_convert before calling pushInput.

Reply:
yes, it is old functions, we can remove it.